### PR TITLE
Update MNK65_STM32

### PR DESCRIPTION
**Pull request by kopibeng**
## Description
Added RGB underglow support and moved to data driven config for MNK65_STM32

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/kopibeng/mnk65_stm32/info.json
Comment (dunk2k) [inline keyboards/kopibeng/mnk65_stm32/info.json:1]: could `qmk format-json` with `-i` please be run on this file?

**Review by dunk2k — CHANGES_REQUESTED** [review]

File: keyboards/kopibeng/mnk65_stm32/info.json [hunk 1]
```diff
@@ -81,22 +978,21 @@
...
+                {"matrix": [4, 10], "x": 10, "y": 4, "w": 1.25},
+                {"matrix": [4, 11], "x": 11.25, "y": 4, "w": 1.25},
                 {"matrix": [4, 12], "x": 13, "y": 4},
                 {"matrix": [4, 13], "x": 14, "y": 4},
                 {"matrix": [4, 14], "x": 15, "y": 4}
             ]
         }
     }
-}
+}
```
Comment (dunk2k) [inline keyboards/kopibeng/mnk65_stm32/info.json:998]: add EoF (it's a GitHub thing)
```suggestion
}

```

**Review by dunk2k — CHANGES_REQUESTED** [review]

File: keyboards/kopibeng/mnk65_stm32/info.json
Comment (kopibeng) [reply keyboards/kopibeng/mnk65_stm32/info.json:1]: done!

**Review by fauxpark — APPROVED** [review]

**Review by dunk2k — APPROVED** [review]

**Review by waffle87 — APPROVED** [review]

File: keyboards/kopibeng/mnk65_stm32/info.json [hunk 2]
```diff
@@ -1,25 +1,925 @@
...
+    "usb": {
+        "device_version": "1.0.0",
+        "pid": "0x0652",
+        "vid": "0x4B50"
+    },
+    "ws2812": {
+        "pin": "B4"
+    },
+    "community_layouts": ["65_ansi_blocker", "65_ansi_blocker_split_bs", "65_ansi_blocker_tsangan", "65_ansi_blocker_tsangan_split_bs", "65_iso_blocker", "65_iso_blocker_split_bs", "65_iso_blocker_tsangan", "65_iso_blocker_tsangan_split_bs"],
     "layouts": {
```
Comment (zvecr) [inline keyboards/kopibeng/mnk65_stm32/info.json:53]: Not backwards compatible without adding an alias, which would then require it to go via `develop`
```suggestion
    "layout_aliases": {"LAYOUT": "LAYOUT_all"},
    "layouts": {
```

**Review by zvecr — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Pull request merged**